From de4cd8d63257af1cb9060c4b2b304a82655adb99 Mon Sep 17 00:00:00 2001
From: bobo1on1 <bob-nospam-@xbmc.org>
Date: Wed, 28 Mar 2012 21:31:46 +0200
Subject: [PATCH] fixed: make sure the old thread is stopped before starting a
 new one, fixes throwing an exception in the main thread

---
 xbmc/peripherals/devices/PeripheralCecAdapter.cpp |   14 ++++++++------
 1 files changed, 8 insertions(+), 6 deletions(-)

diff --git a/xbmc/peripherals/devices/PeripheralCecAdapter.cpp b/xbmc/peripherals/devices/PeripheralCecAdapter.cpp
index 1c91341..a898a16 100644
--- a/xbmc/peripherals/devices/PeripheralCecAdapter.cpp
+++ b/xbmc/peripherals/devices/PeripheralCecAdapter.cpp
@@ -148,15 +148,17 @@ void CPeripheralCecAdapter::Announce(AnnouncementFlag flag, const char *sender,
   }
   else if (flag == System && !strcmp(sender, "xbmc") && !strcmp(message, "OnWake"))
   {
-    // reconnect to the device
-    CSingleLock lock(m_critSection);
-    CLog::Log(LOGDEBUG, "%s - reconnecting to the CEC adapter after standby mode", __FUNCTION__);
+    {
+      // reconnect to the device
+      CSingleLock lock(m_critSection);
+      CLog::Log(LOGDEBUG, "%s - reconnecting to the CEC adapter after standby mode", __FUNCTION__);
 
-    // close the previous connection
-    m_cecAdapter->Close();
+      // close the previous connection
+      m_cecAdapter->Close();
+    }
 
     // and open a new one
-    m_bStop = false;
+    StopThread();
     Create();
   }
 }
-- 
1.7.5.4

